### ISSUE TITLE
Document self-deployments

### ISSUE THREAD

**Issue by brokad** [shuttle-hq/shuttle#81]
The entire platform, client and server code, is open-source and its code is in this repo. However, self-deployment is not documented anywhere so it makes it hard for someone to implement self-deployments by themselves without requiring help from Discord.

### What we can do
Document, as a "getting started" step in the `shuttle-service` crate and in the `README.md` here, that self-deployment is an option and point to a small guide (probably another `.md` file in the root of the repo or a wiki page) which explains how to self-deploy.

**Comment by tv42** [comment]
Please also add an explicit license. Right now Cargo.toml's say `license = "Apache-2.0"` (https://github.com/shuttle-hq/shuttle/search?q=license) but that isn't good enough for lawyers.

**Comment by chesedo** [comment]
The [documentation in the Terraform module](https://github.com/shuttle-hq/shuttle/tree/main/terraform/modules/shuttle) should be enough to get anyone started for now.

The module will be published to Terraform with proper documentation in the near future :smile:

**Comment by stappersg** [comment]
What is the succesor of [documentation in the Terraform module](https://github.com/shuttle-hq/shuttle/tree/main/terraform/modules/shuttle)?  ( today, 2023-01-14, it was responding with _"404 Not Found"_)

**Comment by joshua-mo-143** [comment]
> What is the succesor of [documentation in the Terraform module](https://github.com/shuttle-hq/shuttle/tree/main/terraform/modules/shuttle)? ( today, 2023-01-14, it was responding with _"404 Not Found"_)

Looks like the Terraform modules were made private - at some point it's planned that the repo will be made public again (once all sanity/safety checking is complete).

Discord thread ref: https://discord.com/channels/803236282088161321/1061219495114453052

**Comment by tennox** [comment]
Hey there :)
I know stuff takes time, but I'd really love to see this happen, as I feel I only want to invest in the ecosystem when I'm sure ai could eject from dependency on the company if necessary.

And I like to self-host the stuff I run 😇

**Comment by oddgrd** [comment]
Hey @tennox! While we realize this is important to a lot of people, we are not able to prioritize the work needed to make it easy to bring your own cloud at the moment, but it is on our long-term roadmap. For more information on the current state of self-hosting, see the last entry of our FAQ: https://docs.shuttle.rs/support/faq

While we did plan to, and we may consider it as an intermediate solution in the future, making our IaC repository (terraform/pulumi/ansible) public has been put on hold. It is currently undergoing major changes, and we are not able to prioritize testing, documenting and maintaining it as a solution for self-hosting yet.
